pkg/controller/pxcbackup/controller.go

```diff
@@ -252,18 +252,27 @@ func (r *ReconcilePerconaXtraDBClusterBackup) Reconcile(request reconcile.Reques
 	return rr, err
 }
 
-func removeS3Finalizer(cr *api.PerconaXtraDBClusterBackup) {
-	filteredFins := make([]string, 0)
-
-	for _, f := range cr.GetFinalizers() {
-		if f == api.FinalizerDeleteS3Backup {
-			continue
+func removeS3Finalizer(cl client.Client, cr *api.PerconaXtraDBClusterBackup) error {
+	finalizers := cr.GetFinalizers()
+	beforeLen := len(finalizers)
+
+	for i := 0; i != len(finalizers); i++ {
+		if finalizers[i] == api.FinalizerDeleteS3Backup {
+			copy(finalizers[:i], finalizers[i+1:])
+			finalizers = finalizers[:len(finalizers)-1]
 		}
+	}
 
-		filteredFins = append(filteredFins, f)
+	if beforeLen == len(finalizers) {
+		return nil
 	}
 
-	cr.SetFinalizers(filteredFins)
+	cr.SetFinalizers(finalizers)
+	if err := cl.Update(context.TODO(), cr); err != nil {
+		return errors.Wrap(err, "remove S3 finalizer")
```

Comment:
IMO it's better to describe the action that failed, rather than copying function's name

Reply:
I just did it to have error like `failed to run finalizers: try to run S3 backup finalizer job: remove S3 finalizer: [...root error...]`

but can change it to what you will suggest

Reply:
in this case `update CR` would be better